chainer/device_resident.py

```diff
@@ -56,28 +56,39 @@ def xp(self):
             return None
         return device.xp
 
-    def to_cpu(self):
-        # type: () -> 'DeviceResident'
+    def to_cpu(
+            self,
+            allow_unchain=None,  # type: tp.Optional[bool]
+    ):
+        # type: (...) -> 'DeviceResident'
         """Copies parameter variables and persistent values to CPU.
 
         This method does not handle non-registered attributes. If some of such
         attributes must be copied to CPU, the link implementation should
         override :meth:`~DeviceResident.device_resident_accept` to do so.
 
+        Args:
+            allow_unchain (bool): If ``True``, always unchains the graph. If
```

Comment:
Is the last spec (`None`) needed? The default can simply be `False`.

Reply:
If the default value is `False`, existing user code will break. That's what we ultimately want to but only after the deprecation phase.

Reply:
How about simply raising a warning instead of an error in the deprecation period?

Reply:
I see that's another optional. Could you explain why that's better? It sounds like the user has to be more aware of when the deprecation ends.

Reply:
Of course affected users should be aware of the change. You don't think so?

Another point is simplicity. It's difficult for users to understand the necessity of the third value.

Reply:
I agree on the simplicity. Let me change it. The deprecation warning is only raised when the graphs are connected so it should be rather rare too and not too disturbing neither.

Reply:
Changed the behavior. Also need to fix the docs.

Reply:
Fixed docs.